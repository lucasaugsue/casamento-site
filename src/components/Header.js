import Menu from '@mui/icons-material/Menu';
import { Collapse, List, ListItemButton, ListItemText } from '@mui/material';
import React from 'react';
import styles from './Header.module.css';

export default function Header({transparent : transparentProps = true}){
    const [scrollPosition, setScrollPosition] = React.useState(0);
    const [transparent, setTransparent] = React.useState(false);
    const [open, setOpen] = React.useState(false)

    const listItens = [
        {id: 1, title: "Vamos nos casar"},
        {id: 2, title: "Confirme sua presença"},
        {id: 3, title: "Localização"},
        {id: 4, title: "Livro de visitas"},
    ]

    const handleScroll = e => {
        setScrollPosition(document.documentElement.scrollTop)
    }

    React.useEffect(() => {
        setTransparent(scrollPosition <= 100)
    }, [scrollPosition])

    React.useEffect(() => {
        window.addEventListener('scroll', handleScroll)
        return () => window.removeEventListener('scroll', handleScroll)
    }, [])

    return <div>
        <div className={
            transparent
            ? styles.basicHeaderTransparent
            : styles.basicHeader
        }>
            <div className={styles.gridContainerLogo}>
                <div className={styles.itemLogo}>
                    <div className={styles.logo}>L</div>
                    <div className={styles.logo}>&</div>
                    <div className={styles.logo}>V</div>
                </div>
                <div/>
                <div 
                    className={styles.itemIcon}
                    onClick={() => setOpen(!open)} 
                >
                    <Menu className={styles.icon}/>
                </div>
            </div>
            <Collapse 
                in={open}
                timeout="auto" 
                unmountOnExit
                className={styles.collapse}  
            >
                <List 
                    component="div" 
                    disablePadding
                    className={styles.list} 
                >
                    {listItens.map((i, index) => 
                    <ListItemButton 
                        sx={{ pl: 3 }}
                        key={`${i.id};;${index}`}
                    >
                        <ListItemText primary={i.title} />
                    </ListItemButton>)}
                </List>
            </Collapse>
        </div>
    </div> 
}
