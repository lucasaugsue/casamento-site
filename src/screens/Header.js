import Menu from '@mui/icons-material/Menu';
import { Collapse, List, ListItemButton, ListItemText } from '@mui/material';
import React from 'react';
import styles from './Header.module.css';

export default function Header({transparent : transparentProps = true}){
    const [scrollPosition, setScrollPosition] = React.useState(0);
    const [transparent, setTransparent] = React.useState(false);
    const [open, setOpen] = React.useState(false)

    const listItens = [
        {id: 1, title: "Início", section: "#inicio"},
        {id: 2, title: "Vamos nos casar", section: "#vamos-casar"},
        {id: 3, title: "Confirme sua presença", section: "#confirmar-presenca"},
        {id: 4, title: "Escreva um recado", section: "#recado"},
        {id: 5, title: "Lista de presentes", section: "#presentes"},
        // {id: 6, title: "Localização", section: "#local"},
    ]
    
    const changeOpen = () => setOpen(!open)

    const functionLetra = () => {
        setOpen(false)
        window.location.href=`/`
    }

    const functionSection = (i) => {
        setOpen(false)
        window.location.href=`/${i.section}`
    }

    React.useEffect(() => {
        setTransparent(scrollPosition <= 100)
    }, [scrollPosition])

    React.useEffect(() => {
        const handleScroll = e => {
            setScrollPosition(document.documentElement.scrollTop)
        }

        if (typeof window !== "undefined") {
            
            window.addEventListener('scroll', handleScroll)
            return () => window.removeEventListener('scroll', handleScroll)
        }
        
    }, []) // empty dependencies array means "run this once on first mount"

    return <div>
        <div className={
            transparent
            ? styles.basicHeaderTransparent
            : styles.basicHeader
        }>
            <div className={styles.gridContainerLogo}>
                <div className={styles.itemLogo}>
                    {"L&V".split("")
                    .map((letra, index) => <div 
                        data-open={open}
                        className={styles.logo}
                        key={`${letra};;${index}`}
                        onClick={() => functionLetra()} 
                    >
                        {letra}
                    </div>)}
                </div>
                <div/>
                <div 
                    data-open={open}
                    data-testid="menu-icon"  
                    className={styles.itemIcon}
                    onClick={() => changeOpen()} 
                >
                    <Menu className={styles.icon}/>
                </div>
            </div>
            <Collapse 
                in={open}
                timeout="auto" 
                unmountOnExit
                className={styles.collapse}  
            >
                <List 
                    component="div" 
                    disablePadding
                    className={styles.list} 
                >
                    {listItens.map((i, index) => 
                    <ListItemButton 
                        sx={{ pl: 3 }}
                        data-open={open}
                        key={`${i.id};;${index}`}
                        onClick={() => functionSection(i)}
                    >
                        <ListItemText primary={i.title} />
                    </ListItemButton>)}
                </List>
            </Collapse>
        </div>
    </div> 
}
