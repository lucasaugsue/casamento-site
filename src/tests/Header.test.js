import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Header from '../screens/Header';

/**
 * @jest-environment jsdom
 */

describe('<Header/>', () => {
	describe('Renderização', () => {
		test('deve renderizar Header corretamente', async () => {
			const tree = renderer.create(<Header />).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});

	describe('Interações', () => {
		beforeEach(() => {
			render(<Header />);
		});

		it('deve iniciar com o menu fechado', () => {
			expect(screen.getByTestId('menu-icon')).toBeInTheDocument();
			expect(screen.getByTestId('menu-icon')).toHaveAttribute('data-open', 'false');
		});

		it('deve abrir o menu ao clicar no ícone do menu', async () => {
			fireEvent.click(screen.getByTestId('menu-icon'));
			await waitFor(() => {
				expect(screen.getByTestId('menu-icon')).toHaveAttribute('data-open', 'true');
			});
		});

		it('deve fechar o menu ao clicar na letra "L"', async () => {
			fireEvent.click(screen.getByTestId('menu-icon'));
			await waitFor(() => {
				fireEvent.click(screen.getByText('L'));
				expect(screen.getByText('L')).toHaveAttribute('data-open', 'false');
			});
		});

		it('deve abrir o menu ao clicar no ícone e fechar ao clicar em "Início"', async () => {
			fireEvent.click(screen.getByTestId('menu-icon'));
			await waitFor(() => {
				expect(screen.getByTestId('menu-icon')).toHaveAttribute('data-open', 'true');
			});

			fireEvent.click(screen.getByText('Início'));
			await waitFor(() => {
				expect(screen.getByTestId('menu-icon')).toHaveAttribute('data-open', 'false');
			});
		});
	});
});
