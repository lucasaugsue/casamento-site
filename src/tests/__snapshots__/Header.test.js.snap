// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Renderização deve renderizar Header corretamente 1`] = `
<div>
  <div
    className="basicHeader"
  >
    <div
      className="gridContainerLogo"
    >
      <div
        className="itemLogo"
      >
        <div
          className="logo"
          data-open={false}
          onClick={[Function]}
        >
          L
        </div>
        <div
          className="logo"
          data-open={false}
          onClick={[Function]}
        >
          &
        </div>
        <div
          className="logo"
          data-open={false}
          onClick={[Function]}
        >
          V
        </div>
      </div>
      <div />
      <div
        className="itemIcon"
        data-open={false}
        data-testid="menu-icon"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
